### ISSUE TITLE
Add more topology changes e2e tests with multiple ES versions

### ISSUE THREAD

**Issue by sebgl** [elastic/cloud-on-k8s#679]
To avoid issues such as #678, we should have more topology changes in our mutation e2e tests. They should cover various Elasticsearch major versions (for now: 6.X and 7.X).

**Comment by charith-elastic** [comment]
If I am not mistaken, the case described in #678 is already covered: https://github.com/elastic/cloud-on-k8s/blob/e6d2a411418c9c2675c3f43d568e7431af1fffb3/operators/test/e2e/es/mutation_test.go#L48

Are there other scenarios that need tests?

**Comment by sebgl** [comment]
Things that come to my mind (may not be worth a test for each of them!):

* mutate ES version and ES settings and resources requests at the same time
* rename a NodeSpec (once StatefulSets impl. is merged)
* upgrade (config or resources change) the single master of a cluster
* mutate hot-warm clusters
